locale.c: is_locale_utf8(): C locale cant be UTF-8

khwilliamson opened the pull request:
This adds a quick check to rule out "C" and "POSIX" as being possible UTF-8 locales.  This shortcuts the more expensive checking needed for non-trivial cases.

jkeenan commented:
> This adds a quick check to rule out "C" and "POSIX" as being possible UTF-8 locales. This shortcuts the more expensive checking needed for non-trivial cases.

This could use a test.

khwilliamson commented:
There is no change in external behavior; only a shortcut from the existing code which arrives at the same answer more lugubriously.  Hence the change is effectively not testable.

khwilliamson committed:
locale.c: is_locale_utf8(): C locale cant be UTF-8

This adds a quick check to rule out "C" and "POSIX" as being possible UTF-8 locales. This shortcuts the more expensive checking needed for non-trivial cases.
